### PR TITLE
feat: in memory preferences

### PR DESCRIPTION
**Pull request by aaron-steinfeld**
## Description
Adds a mechanism to store preferences in memory - for just the current application session.

### CODE REVIEW

**Review by palbizu — APPROVED** [review]
Looks well. I will wait for @anandtiwary or @jake-bassett's approval.

**Review by anandtiwary — APPROVED** [review]

**Pull request merged**
